helix-term/src/ui/mod.rs

```diff
@@ -254,11 +245,37 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
         }
     })
     .with_preview(|_editor, path| Some((path.as_path().into(), None)));
-    let injector = picker.injector();
-    let timeout = std::time::Instant::now() + std::time::Duration::from_millis(30);
+    inject_files(picker.injector(), walk_builder, config.file_picker.sorted);
+
+    picker
+}
+
+fn inject_files(
+    injector: Injector<PathBuf, PathBuf>,
+    mut walk_builder: ignore::WalkBuilder,
+    sorted: bool,
+) {
+    use ignore::WalkState;
+    use std::collections::HashSet;
+    use std::time::{Duration, Instant};
+
+    // How long we are willing to wait for results before displaying the file picker.
+    let now = Instant::now();
+    let timeout = now + Duration::from_millis(30);
```

Comment:
This would probably be a config option as well. Otherwise, it's unclear how to reliably trigger all code paths from the integration tests.